bug: panic when using type conversion in a lambda function:

query: 
```
from(bucket: "telegraf")
  |> range(start: -100d)
  |> filter(fn: (r) => r._measurement == "processes")
  |> filter(fn: (r) => r._field == "running" or r._field == "total")
  |> filter(fn: (r) => r.host == "ip-192-168-0-127.ec2.internal")
  |> influxFieldsAsCols()
  |> map(fn: (r) => float(v:r.total), mergeKey: true)
```

trace: 
```

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0x119bf2d]

goroutine 1 [running]:
github.com/influxdata/flux/semantic.(*CallExpression).Copy(0xc0001b3b20, 0xc00015f0e0, 0xc0003c2d01)
	/Users/adam/programming/go/pkg/mod/github.com/influxdata/flux@v0.16.1/semantic/graph.go:635 +0x10d
github.com/influxdata/flux/semantic.(*FunctionBlock).Copy(0xc0002d0000, 0xc0002d8f50, 0xc000017380)
	/Users/adam/programming/go/pkg/mod/github.com/influxdata/flux@v0.16.1/semantic/graph.go:537 +0x8a
github.com/influxdata/flux/semantic.(*FunctionExpression).Copy(0xc00017cc80, 0xc0003c2e28, 0x1c6d3a0)
	/Users/adam/programming/go/pkg/mod/github.com/influxdata/flux@v0.16.1/semantic/graph.go:516 +0x89
github.com/influxdata/flux/interpreter.function.Resolve(0xc00017cc80, 0x1c69260, 0xc0005b03c0, 0xc000016660, 0xc000016690, 0x0, 0x19ffe00, 0x1b167e0, 0xc000016601, 0x2ef6bf0)
	/Users/adam/programming/go/pkg/mod/github.com/influxdata/flux@v0.16.1/interpreter/interpreter.go:705 +0x2f
github.com/influxdata/flux/interpreter.ResolveFunction(0x1c6d3a0, 0xc000017380, 0x2, 0x1c6d3a0, 0xc000017380)
	/Users/adam/programming/go/pkg/mod/github.com/influxdata/flux@v0.16.1/interpreter/interpreter.go:692 +0x67
github.com/influxdata/flux/stdlib/universe.createMapOpSpec(0x1c6d6c0, 0xc000454b60, 0xc00029e990, 0xc0003c2fb0, 0x100e388, 0x20, 0x1a408c0)
	/Users/adam/programming/go/pkg/mod/github.com/influxdata/flux@v0.16.1/stdlib/universe/map.go:54 +0xc4
github.com/influxdata/flux.(*function).call(0xc000161770, 0x1c6d6c0, 0xc000454b60, 0x11f5ef5, 0x1c6df00, 0xc000016600, 0xc000454b60)
	/Users/adam/programming/go/pkg/mod/github.com/influxdata/flux@v0.16.1/compile.go:771 +0x58
github.com/influxdata/flux.(*function).call-fm(0x1c6d6c0, 0xc000454b60, 0x1c6d6c0, 0xc000454b60, 0x30, 0x30)
	/Users/adam/programming/go/pkg/mod/github.com/influxdata/flux@v0.16.1/compile.go:765 +0x3e
github.com/influxdata/flux/interpreter.DoFunctionCall(0xc0003c30e8, 0x1c6df00, 0xc000016600, 0xc0003c30f8, 0x100e388, 0x30, 0x1b167e0)
	/Users/adam/programming/go/pkg/mod/github.com/influxdata/flux@v0.16.1/interpreter/interpreter.go:432 +0x77
github.com/influxdata/flux.(*function).Call(0xc000161770, 0x1c6df00, 0xc000016600, 0xc0005b03c0, 0x1b31e96, 0x6, 0x1c63a40)
	/Users/adam/programming/go/pkg/mod/github.com/influxdata/flux@v0.16.1/compile.go:765 +0x61
github.com/influxdata/flux/interpreter.(*Interpreter).doCall(0xc00057c1c0, 0xc0001b3b90, 0x1c69260, 0xc0005b03c0, 0x0, 0x0, 0x0, 0x1c60340)
	/Users/adam/programming/go/pkg/mod/github.com/influxdata/flux@v0.16.1/interpreter/interpreter.go:477 +0x3d1
github.com/influxdata/flux/interpreter.(*Interpreter).doExpression(0xc00057c1c0, 0x1c63a40, 0xc0001b3b90, 0x1c69260, 0xc0005b03c0, 0x4, 0x1b2f760, 0xc00057c1e8, 0x0)
	/Users/adam/programming/go/pkg/mod/github.com/influxdata/flux@v0.16.1/interpreter/interpreter.go:229 +0x192c
github.com/influxdata/flux/interpreter.(*Interpreter).doStatement(0xc00057c1c0, 0x1c63aa0, 0xc0002d8b90, 0x1c69260, 0xc0005b03c0, 0x100d673, 0x2d5abd8, 0x1, 0x203000)
	/Users/adam/programming/go/pkg/mod/github.com/influxdata/flux@v0.16.1/interpreter/interpreter.go:161 +0x49e
github.com/influxdata/flux/interpreter.(*Interpreter).doFile(0xc00057c1c0, 0xc00016a880, 0x1c69260, 0xc0005b03c0, 0x1c53b00, 0xc00000e188, 0x8, 0xc00000e120)
	/Users/adam/programming/go/pkg/mod/github.com/influxdata/flux@v0.16.1/interpreter/interpreter.go:102 +0x1a0
github.com/influxdata/flux/interpreter.(*Interpreter).doPackage(0xc00057c1c0, 0xc0001b3ab0, 0x1c69260, 0xc0005b03c0, 0x1c53b00, 0xc00000e188, 0x1c53e00, 0xc00066c008)
	/Users/adam/programming/go/pkg/mod/github.com/influxdata/flux@v0.16.1/interpreter/interpreter.go:85 +0xa1
github.com/influxdata/flux/interpreter.(*Interpreter).doRoot(0xc00057c1c0, 0x1c60340, 0xc0001b3ab0, 0x1c69260, 0xc0005b03c0, 0x1c53b00, 0xc00000e188, 0x0, 0x0)
	/Users/adam/programming/go/pkg/mod/github.com/influxdata/flux@v0.16.1/interpreter/interpreter.go:68 +0x154
github.com/influxdata/flux/interpreter.(*Interpreter).Eval(0xc00057c1c0, 0x1c60340, 0xc0001b3ab0, 0x1c69260, 0xc0005b03c0, 0x1c53b00, 0xc00000e188, 0x0, 0x0, 0x0, ...)
	/Users/adam/programming/go/pkg/mod/github.com/influxdata/flux@v0.16.1/interpreter/interpreter.go:59 +0x301
github.com/influxdata/flux/repl.(*REPL).executeLine(0xc00057c200, 0xc0004d6000, 0x13e, 0x0, 0xc0002e19d0, 0x18985e8, 0xc000398000)
	/Users/adam/programming/go/pkg/mod/github.com/influxdata/flux@v0.16.1/repl/repl.go:158 +0x12b
github.com/influxdata/flux/repl.(*REPL).input(0xc00057c200, 0xc0004d6000, 0x13e)
	/Users/adam/programming/go/pkg/mod/github.com/influxdata/flux@v0.16.1/repl/repl.go:123 +0x49
github.com/influxdata/flux/repl.(*REPL).input-fm(0xc0004d6000, 0x13e)
	/Users/adam/programming/go/pkg/mod/github.com/influxdata/flux@v0.16.1/repl/repl.go:51 +0x3e
github.com/c-bata/go-prompt.(*Prompt).Run(0xc000140000)
	/Users/adam/programming/go/pkg/mod/github.com/c-bata/go-prompt@v0.2.2/prompt.go:81 +0x63f
github.com/influxdata/flux/repl.(*REPL).Run(0xc00057c200)
	/Users/adam/programming/go/pkg/mod/github.com/influxdata/flux@v0.16.1/repl/repl.go:63 +0x1d9
main.replF(0x2402ae0, 0xc0005bb1c0, 0x0, 0x2)
	/Users/adam/programming/go/src/github.com/influxdata/influxdb/cmd/influx/repl.go:86 +0x156
github.com/spf13/cobra.(*Command).execute(0x2402ae0, 0xc0005bb120, 0x2, 0x2, 0x2402ae0, 0xc0005bb120)
	/Users/adam/programming/go/pkg/mod/github.com/spf13/cobra@v0.0.3/command.go:766 +0x2cc
github.com/spf13/cobra.(*Command).ExecuteC(0x2402160, 0x1b3473a, 0x9, 0x1b35fd1)
	/Users/adam/programming/go/pkg/mod/github.com/spf13/cobra@v0.0.3/command.go:852 +0x2fd
github.com/spf13/cobra.(*Command).Execute(0x2402160, 0x1007500, 0xc0000a4058)
	/Users/adam/programming/go/pkg/mod/github.com/spf13/cobra@v0.0.3/command.go:800 +0x2b
main.Execute()
	/Users/adam/programming/go/src/github.com/influxdata/influxdb/cmd/influx/main.go:104 +0x2d
main.main()
	/Users/adam/programming/go/src/github.com/influxdata/influxdb/cmd/influx/main.go:15 +0x20

```

--- a/semantic/graph.go
+++ b/semantic/graph.go
@@ -632,7 +632,9 @@ func (e *CallExpression) Copy() Node {
 
 	ne.Callee = e.Callee.Copy().(Expression)
 	ne.Arguments = e.Arguments.Copy().(*ObjectExpression)
-	ne.Pipe = e.Pipe.Copy().(Expression)
+	if e.Pipe != nil {
+		ne.Pipe = e.Pipe.Copy().(Expression)
+	}
 
 	return ne
 }

--- a/stdlib/testing/testdata/map.flux
+++ b/stdlib/testing/testdata/map.flux
@@ -1,0 +1,34 @@
+import "testing"
+
+option now = () => 2030-01-01T00:00:00Z
+
+inData = "
+#datatype,string,long,dateTime:RFC3339,long,string,string,string
+#group,false,false,false,false,true,true,true
+#default,_result,,,,,,
+,result,table,_time,_value,_field,_measurement,host
+,,0,2018-05-22T19:53:26Z,100,load1,system,host.local
+,,0,2018-05-22T19:53:36Z,101,load1,system,host.local
+,,0,2018-05-22T19:53:46Z,102,load1,system,host.local
+"
+
+outData = "
+#datatype,string,long,string,string,string,double
+#group,false,false,true,true,true,false
+#default,_result,,,,,
+,result,table,_field,_measurement,host,newValue
+,,0,load1,system,host.local,100.0
+,,0,load1,system,host.local,101.0
+,,0,load1,system,host.local,102.0
+"
+
+t_map = (table=<-) =>
+  table
+  |> map(fn: (r) => ({newValue: float(v: r._value)}))
+
+testing.test(name: "map",
+            input: testing.loadStorage(csv: inData),
+            want: testing.loadMem(csv: outData),
+            testFn: t_map)
+
+

